src/yz_kv.erl

```diff
@@ -190,8 +190,20 @@ has_indexes(RemoteNode) ->
         _ -> false
     end.
 
-index(Obj, Reason, P) ->
-    case yokozuna:is_enabled(index) andalso ?YZ_ENABLED of
+-spec is_enabled() -> boolean().
+is_enabled() ->
+    yokozuna:is_enabled(index) andalso ?YZ_ENABLED.
+
+index(B, K, Obj, Reason, P) when is_binary(Obj) ->
+    case is_enabled() of
+        true ->
+            RObj = riak_object:from_binary(B, K, Obj),
+            index(B, K, RObj, Reason, P);
+        _ ->
+            ok
+    end;
+index(_B, _K, Obj, Reason, P) ->
+    case is_enabled() of
         true ->
             Ring = yz_misc:get_ring(transformed),
```

Comment:
After looking through the code in a walkthrough (esp. the pars the PR doesn't show, haha), @fadushin and I realized that the `should_index` check should be called before converting the object from binary. This led to a deeper look into some code duplication (around BKey creating, etc...) and whatnot that is happening; so, he's going to be refactoring his changes a bit and just adding a separate index/5 function for binary objects. 

@fre